pmm-app/src/pmm-add-instance/components/AddRemoteInstance/AddRemoteInstance.service.tsx

```diff
@@ -0,0 +1,98 @@
+import { apiRequestManagement } from 'shared/components/helpers/api';
+import { DATABASE_LABELS, Databases } from 'shared/core';
+import { TrackingOptions } from './AddRemoteInstance.types';
+
+class AddRemoteInstanceService {
+  static async addMysql(body) {
+    return apiRequestManagement.post<any, any>('/MySQL/Add', body);
+  }
+
+  static async addPostgresql(body) {
+    return apiRequestManagement.post<any, any>('/PostgreSQL/Add', body);
+  }
+
+  static async addProxysql(body) {
+    return apiRequestManagement.post<any, any>('/ProxySQL/Add', body);
+  }
+
+  static async addMongodb(body) {
+    return apiRequestManagement.post<any, any>('/MongoDB/Add', body);
+  }
+
+  static async addRDS(body) {
+    return apiRequestManagement.post<any, any>('/RDS/Add', body);
+  }
+
+  static addRemote(type, data) {
+    switch (type) {
+      case DATABASE_LABELS[Databases.mongodb]:
+        return AddRemoteInstanceService.addMongodb(data);
+      case DATABASE_LABELS[Databases.mysql]:
+        return AddRemoteInstanceService.addMysql(data);
+      case DATABASE_LABELS[Databases.postgresql]:
+        return AddRemoteInstanceService.addPostgresql(data);
+      case DATABASE_LABELS[Databases.proxysql]:
+        return AddRemoteInstanceService.addProxysql(data);
+      default:
+        throw new Error('Unknown instance type');
+    }
```

Comment:
Switch on type is a code smell (https://wiki.c2.com/?SwitchStatementsSmell) I'd suggest to use a data structure mapping those DB to their type and do something like `new DBs[type]`

Reply:
As discussed in slack, next week we're going to discuss the possibility of restriction switch-case completely